test/integration/etcd/cve_2021_29923_test.go

```diff
@@ -0,0 +1,202 @@
+			ExpectedEtcdPath: "/registry/pods/" + testNamespace + "/pod1",
+		},
+		gvr("", "v1", "services"): {
+			Stub:             `{"metadata": {"name": "service1"}, "spec": {"clusterIP": "10.0.0.011", "externalIP": "192.168.0.012", "externalName": "service1name", "ports": [{"port": 10000, "targetPort": 11000}], "selector": {"test": "data"}}}`,
+			ExpectedEtcdPath: "/registry/services/specs/" + testNamespace + "/service1",
+		},
+		gvr("", "v1", "endpoints"): {
+			Stub:             `{"metadata": {"name": "ep1name"}, "subsets": [{"addresses": [{"hostname": "bar-001", "ip": "192.168.3.011"}], "ports": [{"port": 8000}]}]}`,
+			ExpectedEtcdPath: "/registry/services/endpoints/" + testNamespace + "/ep1name",
+		},
+		// k8s.io/kubernetes/pkg/apis/discovery/v1
+		gvr("discovery.k8s.io", "v1", "endpointslices"): {
+			Stub:             `{"metadata": {"name": "slicev1"}, "addressType": "IPv4", "protocol": "TCP", "ports": [], "endpoints": [{"addresses": ["10.244.0.011"], "conditions": {"ready": true, "serving": true, "terminating": false}, "nodeName": "control-plane"}]}`,
+			ExpectedEtcdPath: "/registry/endpointslices/" + testNamespace + "/slicev1",
+			ExpectedGVK:      gvkP("discovery.k8s.io", "v1beta1", "EndpointSlice"),
+		},
+		// k8s.io/kubernetes/pkg/apis/networking/v1
+		gvr("networking.k8s.io", "v1", "ingresses"): {
+			Stub:             `{"metadata": {"name": "ingress3"}, "spec": {"defaultBackend": {"service":{"name":"service", "port":{"number": 5000}}}}, "status":{"loadBalancer":{"ingress": [{"ip":"10.0.0.013"}]}}}`,
+			ExpectedEtcdPath: "/registry/ingress/" + testNamespace + "/ingress3",
+		},
+		gvr("networking.k8s.io", "v1", "networkpolicies"): {
+			Stub:             `{"metadata": {"name": "np2"}, "spec": {"egress":[{"ports":[{"port":5978,"protocol":"TCP"}],"to":[{"ipBlock":{"cidr":"10.0.012.0/24"}}]}],"ingress":[{"from":[{"ipBlock":{"cidr":"172.017.0.0/16","except":["172.17.001.0/24"]}},{"podSelector":{"matchLabels":{"role":"frontend"}}}],"ports":[{"port":6379,"protocol":"TCP"}]}],"podSelector":{"matchLabels":{"role":"db"}},"policyTypes":["Ingress","Egress"]}}`,
+			ExpectedEtcdPath: "/registry/networkpolicies/" + testNamespace + "/np2",
+		},
+	}
+
+	kindSeen := sets.NewString()
+	pathSeen := map[string][]schema.GroupVersionResource{}
+	etcdSeen := map[schema.GroupVersionResource]empty{}
+	cohabitatingResources := map[string]map[schema.GroupVersionKind]empty{}
+
+	for _, resourceToPersist := range apiServer.Resources {
+		if _, ok := etcdStorageData[resourceToPersist.Mapping.Resource]; !ok {
+			continue
+		}
+		t.Run(resourceToPersist.Mapping.Resource.String(), func(t *testing.T) {
+			mapping := resourceToPersist.Mapping
+			gvk := resourceToPersist.Mapping.GroupVersionKind
+			gvResource := resourceToPersist.Mapping.Resource
+			kind := gvk.Kind
+
+			if kindAllowList.Has(kind) {
```

Comment:
what is kindAllowList ?

Reply:
I've copied the test https://github.com/kubernetes/kubernetes/blob/62503f254eb24d053b7ca0d838cbb1876bfbf5f1/test/integration/etcd/etcd_storage_path_test.go#L72 and removed the things I knew weren't usefult, but there still other "things" I have doubts about them ... I don't know why this global exist, maybe is something legacy, I will remove it